email_backend/data_management/generate_csv.js

```diff
@@ -6,86 +6,106 @@ import fs from "fs";
 import mongoose from "mongoose";
 import { Parser } from "json2csv";
 
-const json2csv_parser = new Parser();
-
 const templates = get_templates();
-connect_db()
-  .then(() =>
-    _.chain(templates)
-      .map((template_value, template_name) => {
-        return make_mongoose_model_from_original_template({
-          original_template: template_value,
-          template_name,
-        });
-      })
-      .map((model) => model.find({}).exec())
-      .thru((promises) => Promise.all(promises))
-      .value()
-  )
-  .then((all_email_logs) => {
-    //if it happens that later on we add another key with a json value,
-    //change reduced json to handle "key.value". For the time being, the CSV
-    //will be cleaner if we flatten the "key.value" to "value"
-    _.forEach(all_email_logs, (collection, index) => {
-      if (collection.length > 0) {
-        const template_name = _.keys(templates)[index];
+const csv_strings = [];
+export const extractor = (testing = false) =>
+  connect_db()
+    .then(() =>
+      _.chain(templates)
+        .map((template_value, template_name) => {
+          return make_mongoose_model_from_original_template({
+            original_template: template_value,
+            template_name,
+          });
+        })
+        .map((model) => model.find({}).exec())
+        .thru((promises) => Promise.all(promises))
+        .value()
+    )
+    .then((all_email_logs) => {
+      //if it happens that later on we add another key with a json value,
+      //change reduced json to handle "key.value". For the time being, the CSV
+      //will be cleaner if we flatten the "key.value" to "value"
+      _.forEach(all_email_logs, (collection, index) => {
+        if (collection.length > 0) {
+          const template_name = _.keys(templates)[index];
 
-        const flattened_email_logs = _.map(collection, function (sub) {
-          return _.chain(
-            _.reduce(
-              sub._doc,
-              function (result, email_log_field, key) {
-                if (key === "__v") {
-                  return result;
-                }
-                if (_.isPlainObject(email_log_field)) {
-                  return { ...result, ...email_log_field };
-                } else if (_.isArray(email_log_field)) {
-                  return { ...result, [key]: email_log_field.join(", ") };
-                } else {
-                  return { ...result, [key]: email_log_field };
-                }
-              },
-              {}
+          const flattened_email_logs = _.map(collection, function (sub, ind) {
+            return _.chain(
+              _.reduce(
+                sub._doc,
+                function (result, email_log_field, key) {
+                  if (key === "__v") {
+                    return result;
+                  }
+                  if (testing && key === "_id") {
+                    return { ...result, [key]: ind };
+                  }
```

Comment:
I think ids might not be necessary in the csv output anyway, just drop them wether or not it's test mode.